packages/oauth/oauth-client/src/oauth-server-agent.ts

```diff
@@ -91,12 +91,18 @@ export class OAuthServerAgent {
     }
   }
 
-  async exchangeCode(code: string, codeVerifier?: string): Promise<TokenSet> {
+  async exchangeCode(
+    code: string,
+    codeVerifier?: string,
+    redirectUri?: string,
```

Comment:
Whilst I cannot see any other usages of `exchangeCode` in the codebase, I marked the `redirectUri` parameter as optional as to avoid any breaking changes in API.